bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductTeaserImpl.java

```diff
@@ -0,0 +1,162 @@
+            }
+            Locale locale = currentPage.getLanguage(false);
+            this.product = getProduct(client, sku);
+            this.priceFormatter = Utils.buildPriceFormatter(locale, this.getCurrency());
+        }
+    }
+
+    @Override
+    public String getName() {
+        return (this.product != null ? this.product.getName() : null);
+    }
+
+    @Override
+    public String getFormattedPrice() {
+        Double price = this.getPrice();
+        if (price != null) {
+            return this.priceFormatter.format(price);
+        }
+        return null;
+    }
+
+    @Override
+    public String getUrl() {
+        return (this.product != null ?  Utils.getUrlfromSlug(productPage.getPath(),product.getUrlKey()) : null);
+    }
+
+    @Override
+    public String getImage() {
+
+        if (this.product != null) {
+            return mediaBaseUrl + PRODUCT_IMAGE_FOLDER + this.product.getSwatchImage();
+        }
+        return null;
+    }
+
+    private String getCurrency() {
+        if (this.product != null) {
+            return this.product.getPrice().getRegularPrice().getAmount().getCurrency().toString();
+        }
+        return null;
+    }
+
+    private Double getPrice() {
+        if (this.product != null) {
+            return this.product.getPrice().getRegularPrice().getAmount().getValue();
+        }
+        return null;
+    }
+
+    // Product DnD from content finder  provides only path of the product to be replaced in
+    private String getSkuFromUrl(String url) {
+       return StringUtils.substringAfterLast(url,"/");
+    }
+
+    private ProductInterface getProduct(GraphqlClient client, String sku) {
+        FilterTypeInput input = new FilterTypeInput().setEq(sku);
+        ProductFilterInput filter = new ProductFilterInput().setSku(input);
+        QueryQuery.ProductsArgumentsDefinition searchArgs = s -> s.filter(filter);
+        ProductsQueryDefinition queryArgs = q -> q.items(this.generateProductQuery());
+        String queryString = Operations.query(query -> query
+                .products(searchArgs, queryArgs)
+                .storeConfig(this.generateStoreConfigQuery())).toString();
+        GraphqlResponse<Query, Error> response = client.execute(new GraphqlRequest(queryString), Query.class, Error.class, QueryDeserializer.getGson());
+        Query rootQuery = response.getData();
+        List<ProductInterface> products = rootQuery.getProducts().getItems();
+        mediaBaseUrl = rootQuery.getStoreConfig().getSecureBaseMediaUrl();
+        if (products.size() > 0) {
+            return products.get(0);
+        }
+        return null;
+    }
+
+    private ProductPricesQueryDefinition generatePriceQuery() {
+        return q -> q
+                .regularPrice( rp -> rp
+                        .amount( a -> a.currency().value()));
+    }
+
+    private ProductInterfaceQueryDefinition generateProductQuery() {
+        return q -> q
+                .name()
+                .swatchImage()
```

Comment:
I know I said use swatch image last time, but maybe better change this to `image`. `image` also returns the full qualified URL, so the store config query part can be removed at all.